packages/docs/getting-started/running-a-validator.md

```diff
@@ -46,33 +66,227 @@ The code you'll see on this page is bash commands and their output.
+export CELO_IMAGE=us.gcr.io/celo-testnet/celo-node
+export NETWORK_ID=1101
 ```
 
-## **Create accounts**
+In all the commands we are going to see the `CELO_IMAGE` and `CELO_NETWORK` variables to refer to the right Docker image and network to use. Now we can get the Docker image:
 
-Create and cd into the directory where you want to store the data and any other files needed to run your node. You can name this whatever you’d like, but here’s a default you can use:
+```bash
+docker pull $CELO_IMAGE:$CELO_NETWORK
+```
+
+#### Create accounts
+
+At this point we need to create the accounts that will be used by the Validator and the Proxy. We create and cd into the directory where you want to store the data and any other files needed to run your node. You can name this whatever you’d like, but here’s a default you can use:
 
 ```bash
-mkdir celo-data-dir
+mkdir -p celo-data-dir/proxy celo-data-dir/validator
 cd celo-data-dir
 ```
 
+We are going to need to create 3 accounts, 2 for the validator and 1 for the Proxy.
+
+First we create three accounts, one for the Validator, one for Validator Group and the last one for the Proxy. You can get their addresses if you don’t already have them. If you already have some accounts, you can skip this step.
+
+To create the accounts needed, run the following commands. The first two create the accounts for the validator, the third one for the proxy:
+
+```bash
+docker run -v $PWD/validator:/root/.celo --entrypoint /bin/sh -it $CELO_IMAGE:$CELO_NETWORK -c "geth account new"
+docker run -v $PWD/validator:/root/.celo --entrypoint /bin/sh -it $CELO_IMAGE:$CELO_NETWORK -c "geth account new"
+docker run -v $PWD/proxy:/root/.celo --entrypoint /bin/sh -it $CELO_IMAGE:$CELO_NETWORK -c "geth account new"
+```
+
+Those commands will prompt you for a passphrase, ask you to confirm it, and then will output your account address: `Address: {<YOUR-ACCOUNT-ADDRESS>}`
+
+Let's save these addresses to environment variables, so that you can reference it later (don't include the braces):
+
+```bash
+export CELO_VALIDATOR_GROUP_ADDRESS=<YOUR-VALIDATOR-GROUP-ADDRESS>
+export CELO_VALIDATOR_ADDRESS=<YOUR-VALIDATOR-ADDRESS>
+export CELO_PROXY_ADDRESS=<YOUR-PROXY-ADDRESS>
+```
+
+In order to register the validator later on, generate a "proof of possession" - a signature proving you know your validator's BLS private key. Run this command:
+
+```bash
+docker run -v $PWD/validator:/root/.celo --entrypoint /bin/sh -it $CELO_IMAGE:$CELO_NETWORK -c "geth account proof-of-possession $CELO_VALIDATOR_ADDRESS"
+```
+
+It will prompt you for the passphrase you've chosen for the validator account. Let's save the resulting proof-of-possession to an environment variable:
+
+```bash
+export CELO_VALIDATOR_POP=<YOUR-VALIDATOR-PROOF-OF-POSSESSION>
+```
+
+#### Deploy the validator and proxy nodes
+
+We initialize the docker containers for the validator and the proxy, building from an image for the network and initializing Celo with the genesis block:
```

Comment:
I could see people being confused where `/celo/genesis.json` came from, so maybe specify "with the genesis block found inside the Docker image"

Reply:
changed